financial-connections/src/main/java/com/stripe/android/financialconnections/features/partnerauth/PartnerAuthViewModel.kt

```diff
@@ -161,16 +161,21 @@ internal class PartnerAuthViewModel @Inject constructor(
 
     private suspend fun completeAuthorizationSession() {
         kotlin.runCatching {
+            setState { copy(authenticationStatus = Loading()) }
             val authSession = requireNotNull(getManifest().activeAuthSession)
-            logger.debug("Web AuthFlow completed! waiting for oauth results")
-            val oAuthResults = pollAuthorizationSessionOAuthResults(authSession)
-            logger.debug("OAuth results received! completing session")
-            val updatedSession = completeAuthorizationSession(
-                authorizationSessionId = authSession.id,
-                publicToken = oAuthResults.memberGuid
-            )
-            logger.debug("Session authorized!")
-            goNext(updatedSession.nextPane)
+            if (authSession.isOAuth) {
+                logger.debug("Web AuthFlow completed! waiting for oauth results")
+                val oAuthResults = pollAuthorizationSessionOAuthResults(authSession)
+                logger.debug("OAuth results received! completing session")
+                val updatedSession = completeAuthorizationSession(
+                    authorizationSessionId = authSession.id,
+                    publicToken = oAuthResults.memberGuid
+                )
+                logger.debug("Session authorized!")
+                goNext(updatedSession.nextPane)
```

Comment:
note that Stripe.js (and therefore iOS) seems to hard code this next pane:
- https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/961f061e/src/linkedAccounts/inner/components/panes/PartnerAuth/index.tsx#L201

not saying its correct, but it MAYBE is dangerous to assume that nextPane is always correct here (probably not, but I guess it wouldn't be extensively tested if I am right)

Reply:
NVM this code changed to t his:
- https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/master/src/linkedAccounts/inner/components/panes/PartnerAuth/index.tsx#L205